index.js

```diff
@@ -168,7 +176,24 @@ class KeyringController extends EventEmitter {
    * @returns {Promise<Object>} A Promise that resolves to the state.
    */
   async submitPassword(password) {
+    await this.verifyPassword(password);
     this.keyrings = await this.unlockKeyrings(password);
+
+    // MV3: If we're provided a password, we should persist keyrings
```

Comment:
This package is used by mobile and MV2 extension as well, and these changes to the workflow will apply to both. This "MV3:" prefix might be confusing or misleading.